src/screens/TokenDetail.js

```diff
@@ -0,0 +1,685 @@
+        'token': this.state.token.uid
+      });
+    }
+    const promise = hathorLib.tokens.destroyAuthority(data, this.state.pin);
+    this.handlePromise(promise, `${label} outputs destroyed!`);
+  }
+
+  /**
+   * Execute destroy mint method
+   */
+  executeDestroyMint = () => {
+    this.executeDestroy(this.state.mintOutputs, 'Mint');
+  }
+  
+  /**
+   * Called when clicking to destroy mint outputs.
+   * Validate if we have the quantity of outputs requested to destroy and open the PIN modal
+   */
+  destroyMint = () => {
+    if (this.destroyQuantity.current.value > this.state.mintOutputs.length) {
+      this.setState({ errorMessage: `You only have ${this.state.mintOutputs.length} mint ${hathorLib.helpers.plural(this.state.mintOutputs.length, 'output', 'outputs')} to destroy.` });
+      return;
+    }
+    this.openPinModal();
+  }
+
+  /**
+   * Execute destroy melt method
+   */
+  executeDestroyMelt = () => {
+    this.executeDestroy(this.state.meltOutputs, 'Melt');
+  }
+
+  /**
+   * Called when clicking to destroy melt outputs.
+   * Validate if we have the quantity of outputs requested to destroy and open the PIN modal
+   */
+  destroyMelt = () => {
+    if (this.destroyQuantity.current.value > this.state.meltOutputs.length) {
+      this.setState({ errorMessage: `You only have ${this.state.meltOutputs.length} melt ${hathorLib.helpers.plural(this.state.meltOutputs.length, 'output', 'outputs')} to destroy.` });
+      return;
+    }
+    this.openPinModal();
+  }
+
+  /**
+   * Execute mint method after form validation
+   */
+  executeMint = () => {
+    const amountValue = this.amount.current.value*(10**hathorLib.constants.DECIMAL_PLACES);
+    const output = this.state.mintOutputs[0];
+    const address = this.chooseAddress.current.checked ? hathorLib.wallet.getAddressToUse() : this.address.current.value;
+    const promise = hathorLib.tokens.mintTokens(output.tx_id, output.index, output.decoded.address, this.state.token.uid, address, amountValue, this.state.pin, this.createAnother.current.checked, false);
+    this.handlePromise(promise, `${hathorLib.helpers.prettyValue(amountValue)} ${this.state.token.symbol} minted!`);
+  }
+
+  /**
+   * Method executed after user clicks on mint button.
+   * Validates the form and then opens the PIN modal
+   */
+  mint = () => {
+    if (this.chooseAddress.current.checked === false && this.address.current.value === '') {
+      this.setState({ errorMessage: 'Address is required when not selected automatically' });
+      return;
+    }
+    this.openPinModal();
+  }
+
+  /**
+   * Execute melt method after form validation
+   */
+  executeMelt = () => {
+    const amountValue = this.amount.current.value*(10**hathorLib.constants.DECIMAL_PLACES);
+    const output = this.state.meltOutputs[0];
+    const promise = hathorLib.tokens.meltTokens(output.tx_id, output.index, output.decoded.address, this.state.token.uid, amountValue, this.state.pin, this.createAnother.current.checked);
```

Comment:
Does it generate and propagate the transaction? Maybe we should split into two methods, one to generate, and another one to generate (calling the first one) and then to propagate. What do you think?

Reply:
It's better if in the future we just want the tx before propagating. Done